provider/cmd/pulumi-gen-kubernetes/main.go

```diff
@@ -319,9 +319,6 @@ func genK8sResourceTypes(pkg *schema.Package) {
 		contract.Assert(len(parts) == 3)
 
 		groupVersion, kind := parts[1], parts[2]
-		if strings.HasSuffix(kind, "List") {
```

Comment:
I realized that we're not handling List deprecations for a few cases. Could you update that logic to include the corresponding List types as well? https://github.com/pulumi/pulumi-kubernetes/pull/1182/files#diff-239b11d00160719500c6fea14202bd9eL150-L157